src/sessions.ts

```diff
@@ -467,10 +467,10 @@ export class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
    * @param fn - A lambda to run within a transaction
    * @param options - Optional settings for the transaction
    */
-  withTransaction<T = void>(
-    fn: WithTransactionCallback<T>,
+  withTransaction(
+    fn: WithTransactionCallback<void>,
     options?: TransactionOptions
-  ): ReturnType<typeof fn> {
+  ): Promise<void> {
```

Comment:
Even though the return type is meant to be an internal command result we should reflect what's actually returned here and we can fix this in the next major version. This function will return the command result if the transaction commits successfully, it will throw if it aborts and it will return undefined if it's aborted manually within the user provided function, so it's important that the result at least demonstrate that it can be truthy or not under the right circumstance. 
So we should put the type in here that matches the actual result from the function, and add tests to prevent a regression.